server/node/src/main/java/org/kaaproject/kaa/server/node/service/initialization/KaaNodeInitializationService.java

```diff
@@ -100,18 +105,23 @@ public void start() {
             LOG.error("Interrupted while waiting for thrift to start...", e);
         }
 
-        if (getNodeConfig().isControlServiceEnabled()) {
-            controlInitializationService.start();
-        }
-        if (getNodeConfig().isBootstrapServiceEnabled()) {
-            bootstrapInitializationService.start();
-        }
-        if (getNodeConfig().isOperationsServiceEnabled()) {
-            operationsInitializationService.start();
+        if (waitZkConnection()) {
+            if (getNodeConfig().isControlServiceEnabled()) {
+                controlInitializationService.start();
+            }
+            if (getNodeConfig().isBootstrapServiceEnabled()) {
+                bootstrapInitializationService.start();
+            }
+            if (getNodeConfig().isOperationsServiceEnabled()) {
+                operationsInitializationService.start();
+            }
+
+            LOG.info("Kaa Node Server Started.");
+        } else {
+            LOG.error("Failed to connect to Zookeeper for {} minutes.", getNodeConfig().getZkWaitConnectionTime());
```

Comment:
Failed to connect to Zookeeper within {} minutes. Kaa Node Server will be stopped.